Auth upgrade failed on follower when replaying journal

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create a 3fe cluster with version 2.5.0
2. restart them simutaneously with new version lib

### Expected behavior (Required)
upgrade success

### Real behavior (Required)
```
2023-02-02 21:38:33,815 WARN (replayer|85) [GlobalStateMgr.replayJournalInner():1961] catch exception when replaying 999,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10267
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:1011) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1950) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1807) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1872) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.mysql.privilege.AuthUpgrader.upgradeUser(AuthUpgrader.java:100) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.privilege.AuthUpgrader.replayUpgrade(AuthUpgrader.java:92) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayAuthUpgrade(GlobalStateMgr.java:3541) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:989) ~[starrocks-fe.jar:?]
        ... 4 more
2023-02-02 21:38:33,820 WARN (replayer|85) [GlobalStateMgr$5.runOneCycle():1810] got interrupt exception or inconsistent exception when replay journal 999, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 10267
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:1011) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1950) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1807) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1872) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.mysql.privilege.AuthUpgrader.upgradeUser(AuthUpgrader.java:100) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.privilege.AuthUpgrader.replayUpgrade(AuthUpgrader.java:92) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayAuthUpgrade(GlobalStateMgr.java:3541) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:989) ~[starrocks-fe.jar:?]
        ... 4 more
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java
@@ -1252,7 +1252,7 @@ public class GlobalStateMgr {
         if (feType == FrontendNodeType.OBSERVER || feType == FrontendNodeType.FOLLOWER) {
             Preconditions.checkState(newType == FrontendNodeType.UNKNOWN);
             LOG.warn("{} to UNKNOWN, still offer read service", feType.name());
-            // not set canRead here, leave canRead as what is was.
+            // not set canRead here, leave canRead as what it was.
             // if meta out of date, canRead will be set to false in replayer thread.
             metaReplayState.setTransferToUnknown();
             feType = newType;
@@ -3537,12 +3537,14 @@ public class GlobalStateMgr {
     }
 
     public void replayAuthUpgrade(AuthUpgradeInfo info) throws AuthUpgrader.AuthUpgradeUnrecoverableException {
-        AuthUpgrader upgrader = new AuthUpgrader(auth, authenticationManager, privilegeManager, this);
-        upgrader.replayUpgrade(info.getRoleNameToId());
-        usingNewPrivilege.set(true);
-        domainResolver.setAuthenticationManager(authenticationManager);
-        if (!isCheckpointThread()) {
-            this.auth = null;
+        if (GlobalStateMgr.getCurrentState().needUpgradedToNewPrivilege()) {
+            AuthUpgrader upgrader = new AuthUpgrader(auth, authenticationManager, privilegeManager, this);
+            upgrader.replayUpgrade(info.getRoleNameToId());
+            usingNewPrivilege.set(true);
+            domainResolver.setAuthenticationManager(authenticationManager);
+            if (!isCheckpointThread()) {
+                this.auth = null;
+            }
         }
     }
 

